extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitter.java

```diff
@@ -170,30 +189,69 @@ public void emit(final Event event)
       Map<String, Object> result = resultBuilder.build();
 
       try {
-        String resultJson = jsonMapper.writeValueAsString(result);
-        ObjectContainer<String> objectContainer = new ObjectContainer<>(
-            resultJson,
-            StringUtils.toUtf8(resultJson).length
+        byte[] resultBytes = jsonMapper.writeValueAsBytes(result);
+        ObjectContainer<byte[]> objectContainer = new ObjectContainer<>(
+            resultBytes,
+            resultBytes.length
         );
         if (event instanceof ServiceMetricEvent) {
-          if (!metricQueue.offer(objectContainer)) {
+          if (config.getMetricTopic() == null || !metricQueue.offer(objectContainer)) {
             metricLost.incrementAndGet();
           }
         } else if (event instanceof AlertEvent) {
-          if (!alertQueue.offer(objectContainer)) {
+          if (config.getAlertTopic() == null || !alertQueue.offer(objectContainer)) {
             alertLost.incrementAndGet();
           }
         } else if (event instanceof RequestLogEvent) {
           if (config.getRequestTopic() == null || !requestQueue.offer(objectContainer)) {
             requestLost.incrementAndGet();
           }
+        } else if (event instanceof SegmentMetadataEvent) {
+          if (config.getSegmentMetadataTopic() != null && config.getSegmentMetadataTopicFormat().equals(KafkaEmitterConfig.PROTOBUF)) {
+            resultBytes = convertMetadataEventToProto((SegmentMetadataEvent) event, segmentMetadataLost);
+            objectContainer = new ObjectContainer<>(
+                resultBytes,
+                resultBytes.length
+            );
+          }
+          if (config.getSegmentMetadataTopic() == null || !segmentMetadataQueue.offer(objectContainer)) {
```

Comment:
can we avoid the second null check here and nest it all under one check to make the code more readable?

Reply:
Updated